features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/CommentAdd.jsx

```diff
@@ -94,27 +99,27 @@ class CommentAdd extends React.Component {
      * @memberof CommentAdd
      */
     handleClickCancel() {
-        this.setState({ content: '' });
+        this.setState({ content: '', replyIndex: -1 });
         // const { toggleShowReply } = this.props;
         // toggleShowReply();
     }
 
-    // /**
-    //  * Filters the comment to add the reply
-    //  * @memberof CommentAdd
-    //  */
-    // filterCommentToAddReply(commentToFilter) {
-    //     const { parentCommentId } = this.props;
-    //     return commentToFilter.commentId === parentCommentId;
-    // }
+    /**
+     * Filters the comment to add the reply
+     * @memberof CommentAdd
+     */
+    filterCommentToAddReply(commentToFilter) {
+        const { parentCommentId } = this.props;
+        return commentToFilter.id === parentCommentId;
+    }
 
     /**
      * Handles adding a new comment
      * @memberof CommentAdd
      */
     handleClickAddComment() {
         const {
-            apiId, allComments, commentsUpdate, intl,
+            apiId, allComments, commentsUpdate, intl, parentCommentId,
         } = this.props;
         const { content } = this.state;
         const Api = new API();
```

Comment:
I think we need to provide a parent comment ID when creating a new comment/reply now